hierarqcal/utils.py

```diff
@@ -144,10 +147,81 @@ def plot_motif(
             )
             ax.add_patch(arrow)
         else:
-            # TODO hyper graph visual (rubber band)
-            raise NotImplementedError(
-                f"Plotting {len(edge)}-ary edges are not implemented yet"
-            )
+            end_points = []
+            for i in range(len(edge)):
+                if i == 0:
+                    color = color_dict['start']
+                elif i == len(edge)-1:
+                    color = color_dict['end']
+                else:
+                    color = color_dict['during']
+                
+                ax.add_patch(
+                    patches.Circle(
+                        pos[edge[i]], radius=node_radi[labels.index(edge[i])], edgecolor="black", facecolor=color, linewidth=1.5
+                    )
+                )
+                
+            for i in range(len(edge)):
+                ket_s = np.array(pos[edge[i]])  # source vector
+                ket_t = np.array(pos[edge[(i+1)%len(edge)]])  # target vector
+
+                par_vec = ket_t-ket_s
+                
+                orth_vec = np.array([par_vec[1], -par_vec[0]])*(1/np.linalg.norm(par_vec))
+                
+                if primitive.arity < primitive.step:
```

Comment:
Can you add a comment here to explain why this is done

Reply:
Sure will add the comment.